### DISCUSSION TITLE
How does ExceptionHandlerHelper return information in a custom format？

### DISCUSSION THREAD

**Discussion by ecfxs** [MarcinOrlowski/laravel-api-response-builder#259] [General]
## Steps to reproduce the behavior: ##

I defined a app/Http/Response/RB.php 
![image](https://github.com/MarcinOrlowski/laravel-api-response-builder/assets/16558562/2b7bcf6c-f5e5-4e87-9e1c-06363c2a664f)

And used ExceptionHandlerHelper
How can I return the error message in the RB format I defined?

This is the default format
![image](https://github.com/MarcinOrlowski/laravel-api-response-builder/assets/16558562/af9c54aa-fe9b-4c76-ba11-7dfe7c360028)

This is not the format I want

Thank you

## Environment ##

* Name and version of OS: [mac os]
* PHP version [i.e. 8.1]
* Laravel version [i.e. 10.12.0]

**Comment by TheCoderRaman** [comment]
### Hi @ecfxs ,

The quick fix for you is to edit `App\Exceptions\Handler.php`

**Incase:** You are not doing exception handling with `Response Builder`
See  [Exception Handling](https://github.com/MarcinOrlowski/laravel-api-response-builder/blob/master/docs/exceptions.md) for reference for using `Response Builder` exception handler.

```php
    /**
     * Register the exception handling callbacks for the application.
     *
     * @return void
     */
    public function register(): void
    {
        $this->renderable(function (Throwable $ex, $request) {
            return ExceptionHandlerHelper::handle($ex, $request);
        });
    }
```

Next, update the register method as follows

```php
    /**
     * Register the exception handling callbacks for the application.
     *
     * @return void
     */
    public function register(): void
    {
        $this->renderable(function (\Throwable $ex, $request) {
            $response = ExceptionHandlerHelper::render($request, $ex);

            // Get response data from json response
            $data = $response->getData(true);

            // Modify the response format as needed
            $date = new \DateTime();
            $data['timezone'] = $date->getTimezone();
            $data['timestamp'] = $date->getTimestamp();

            // Update the json response with new data
            $response->setData($data);

            return $response;
        });
    }
```

#### The process involves 
Rendering the response using `ExceptionHandlerHelper` followed by 
modifying the rendered response data as necessary, and subsequently updating the response with the new data.
